### PR TITLE
pass authKeys to MTProto

### PR DESCRIPTION
**Pull request by nodatall**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -12,7 +12,6 @@
 /build
 /dist
 /package
-/scheme/index.js
```
Comment (jaredatron) [inline .gitignore:15]: i thought we were going to stop pulling in this file

File: package.json [hunk 2]
```diff
@@ -27,6 +27,7 @@
   "files": [
     "index.js",
     "src/**/*.js",
+    "scripts/*.js",
```
Comment (jaredatron) [inline package.json:30]: i thought we were going to add a post-install hook to get that large dataset

File: src/index.js [hunk 3]
```diff
@@ -53,20 +53,23 @@ const PRODUCTION_DC_LIST = [
 
 class MTProto {
   constructor(options) {
-    const { api_id, api_hash, test = false, customLocalStorage } = options;
+    const { api_id, api_hash, test = false, authKeys } = options;
 
     this.api_id = api_id;
     this.api_hash = api_hash;
 
     this.dcList = test ? TEST_DC_LIST : PRODUCTION_DC_LIST;
```
Comment (jaredatron) [inline src/index.js]:
```suggestion
    this.dcList = MTProto.dcList(test);
```

Comment (jaredatron) [reply src/index.js]: OR
```suggestion
    this.dcList = this.constructor.dcList(test);
```

File: src/index.js [hunk 4]
```diff
@@ -146,10 +149,38 @@ class MTProto {
...
+    })
+    return JSON.stringify(authKeys);
+  }
+}
+
+MTProto.dcList = function(test = false) {
+  return test ? TEST_DC_LIST : PRODUCTION_DC_LIST;
+}
+
+MTProto.authStorageKeys = function(test = false){
```
Comment (jaredatron) [inline src/index.js]: I'm still confused by the dual meaning of keys in this code.

File: src/storage/index.js [hunk 5]
```diff
@@ -1,12 +1,10 @@
...
 const cache = {};
 
 class Storage {
   constructor(prefix, options = {}) {
     this._prefix = prefix;
 
-    const { customLocalStorage = localStorage } = options;
+    const { customLocalStorage } = options;
 
     this.localStorage = customLocalStorage;
```
Comment (jaredatron) [inline src/storage/index.js]: why do we need `this.localStorage` if you removed it being used in `set` and `get`?

**Review by jaredatron — COMMENTED** [review]
is ba231700089616cfe402d59135c40ca3fc497761 really your complete set of modifications to mtproto-core?

**Review by jaredatron — CHANGES_REQUESTED** [review]
questions

File: .gitignore (hunk 1 above)
Comment (nodatall) [reply .gitignore:15]: Nope, it turns out the dataset is made from two even bigger sets that would have to be included to get the post install script to work, and this file is in the package when you get it from npm

File: package.json (hunk 2 above)
Comment (nodatall) [reply package.json:30]: see above comment

File: src/index.js (hunk 4 above)
Comment (nodatall) [reply src/index.js]: renamed to `authStorageProps`

File: src/storage/index.js (hunk 5 above)
Comment (nodatall) [reply src/storage/index.js]: we don't, removed

**Pull request merged**